tests/integration/ha/test_backups.py

```diff
@@ -0,0 +1,570 @@
+    ops_test: OpsTest,
+    cloud_configs: Dict[str, Dict[str, str]],
+    cloud_credentials: Dict[str, Dict[str, str]],
+    cloud_name: str,
+) -> None:
+    """Deletes the entire OpenSearch cluster and redeploys from scratch.
+
+    Restores each of the previous backups we created and compare with their doc count.
+    The cluster is considered healthy if:
+    1) At each backup restored, check our track of doc count vs. current index count
+    2) Try to write to that new index.
+    """
+    if app := await app_name(ops_test):
+        return
+    logging.info("Destroying the application")
+    await asyncio.gather(
+        ops_test.model.remove_application(S3_INTEGRATOR, block_until_done=True),
+        ops_test.model.remove_application(app, block_until_done=True),
+        ops_test.model.remove_application(TLS_CERTIFICATES_APP_NAME, block_until_done=True),
+    )
+
+    logging.info("Deploying a new cluster")
+    my_charm = await ops_test.build_charm(".")
+    await ops_test.model.set_config(MODEL_CONFIG)
+    # Deploy TLS Certificates operator.
+    config = {"ca-common-name": "CN_CA"}
+
+    await asyncio.gather(
+        ops_test.model.deploy(TLS_CERTIFICATES_APP_NAME, channel="stable", config=config),
+        ops_test.model.deploy(S3_INTEGRATOR, channel=S3_INTEGRATOR_CHANNEL),
+        ops_test.model.deploy(my_charm, num_units=3, series=SERIES),
+    )
+
+    # Relate it to OpenSearch to set up TLS.
+    await ops_test.model.integrate(APP_NAME, TLS_CERTIFICATES_APP_NAME)
+    await ops_test.model.wait_for_idle(
+        apps=[TLS_CERTIFICATES_APP_NAME, APP_NAME],
+        status="active",
+        timeout=1400,
+        idle_period=IDLE_PERIOD,
+    )
+    # Credentials not set yet, this will move the opensearch to blocked state
+    # Credentials are set per test scenario
+    await ops_test.model.integrate(APP_NAME, S3_INTEGRATOR)
+
+    leader_id = await get_leader_unit_id(ops_test)
+    unit_ip = await get_leader_unit_ip(ops_test)
+    config: Dict[str, str] = cloud_configs[cloud_name]
+
+    logger.info(f"Syncing credentials for {cloud_name}")
+    await _configure_s3(ops_test, config, cloud_credentials[cloud_name], app)
+    backups = await list_backups(ops_test, leader_id)
+
+    global cwrites_backup_doc_count
+    # We are expecting 2x backups available
+    assert len(backups) == 2
+    assert len(cwrites_backup_doc_count) == 2
+    count = 0
+    for backup_id in backups.keys():
+        assert await restore(ops_test, backup_id, unit_ip, leader_id)
+        count = await index_docs_count(ops_test, app, unit_ip, ContinuousWrites.INDEX_NAME)
+
+        # Ensure we have the same doc count as we had on the original cluster
+        assert count == cwrites_backup_doc_count[backup_id]
+
+        # restart the continuous writes and check the cluster is still accessible post restore
+        assert await start_and_check_continuous_writes(ops_test, unit_ip, app)
+
+    # Now, try a backup & restore with continuous writes
```

Comment:
where is the restore?

Reply:
Indeed, there is a method to do so, but its name was not really helpful! I renamed it to `assert_restore_indices_and_compare_consistency` so one can clearly see this check